flume-ng-tests/src/test/java/org/apache/flume/test/util/DockerInstall.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.flume.test.util;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Files;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Attempts to setup a staged install using explicitly specified tar-ball
+ * distribution or by using relative path into the flume-ng-dist module.
+ */
+public class DockerInstall extends StagedInstall {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DockerInstall.class);
+
+  private String configFilePath;
+
+  private ProcessShutdownHook shutdownHook;
+  private ProcessInputStreamConsumer consumer;
+  private String containerId;
+
+  private static DockerInstall INSTANCE;
+  private String dockerImageId;
+
+  public static synchronized DockerInstall getInstance() throws Exception {
+    if (INSTANCE == null) {
+      INSTANCE = new DockerInstall();
+    }
+    return INSTANCE;
+  }
+
+  public synchronized boolean isRunning() {
+    return containerId != null;
+  }
+
+  public synchronized void stopAgent() throws Exception {
+    if (containerId == null) {
+      throw new Exception("Process not found");
+    }
+
+    LOGGER.info("Shutting down agent process");
+
+    ImmutableList.Builder<String> builder = new ImmutableList.Builder<>();
+    builder.add("/bin/sh");
+    builder.add("-c");
+    builder.add("'docker kill " +  containerId + "'");
```

Comment:
all of the cases where docker is invoked here should be an environment variable or some such that defaults to "docker" and allows whoever is running the test to specify a fully qualified path to the docker executable.